rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/NodeSource.java

```diff
@@ -118,6 +118,25 @@ public int hashCode() {
         return sourceName != null ? sourceName.hashCode() : 0;
     }
 
+    public boolean hasTheSameParams(Object o) {
+        if (o == null || getClass() != o.getClass())
+            return false;
+        NodeSource nodeSource = (NodeSource) o;
+
+        if (this.getSourceName().equals(nodeSource.getSourceName()) &&
```

Comment:
this big if test should be refactored in a method

boolean checkNodeSourceDetailsEquals(NodeSource other)

Reply:
I have renamed the methods. 
For the tests, we agreed that they are not worth it as there would be a lot of code to test every possible scenario